src/extension/css/src/main/java/org/geoserver/community/css/web/StylePanel.java

```diff
@@ -23,19 +23,24 @@
 import org.apache.wicket.model.IModel;
 import org.apache.wicket.model.PropertyModel;
 import org.geoserver.catalog.StyleInfo;
+import org.geoserver.platform.resource.Resource;
+import org.geoserver.platform.resource.Resources;
 import org.geoserver.web.wicket.CodeMirrorEditor;
 
 public class StylePanel extends Panel {
     private String styleBody;
 
     public StylePanel(String id, IModel<CssDemoPage> model, final CssDemoPage page,
-            final File cssFile) {
+            final Resource cssFile) {
         super(id, model);
-        if (cssFile != null && cssFile.exists()) {
+        if (cssFile != null && Resources.exists(cssFile)) {
+            InputStream is = cssFile.in();
             try {
-                styleBody = FileUtils.readFileToString(cssFile);
+                styleBody = IOUtils.toString(is, "UTF-8");
```

Comment:
On 01/12/15 03:32, NielsCharlier wrote:
> No, but I can get rid of that second parameter and then it will use the platform default.

Niels, I think that would be worse. The only solution I can think of is 
to determine the encoding first, but CSS does not require the encoding 
to be self-documented. Mozilla uses UTF-8 only as a fallback when there 
is no BOM or @charset:
https://developer.mozilla.org/en/docs/Web/CSS/@charset

Kind regards,

-- 
Ben Caradoc-Davies <ben@transient.nz>
Director
Transient Software Limited <http://transient.nz/>
New Zealand